src/libs/auth.ts

```diff
@@ -97,49 +100,68 @@ const revokeTokens = async () => {
 const getSigninArgs = (includeBillingScope) => {
   return Utils.cond(
     [includeBillingScope === false, () => ({})],
-    // For Google just append the scope to the signin args.
-    [isGoogleAuthority(), () => ({ scope: 'openid email profile https://www.googleapis.com/auth/cloud-billing' })],
-    // For B2C switch to a dedicated policy endpoint configured for the GCP cloud-billing scope.
+    // For B2C use a dedicated policy endpoint configured for the GCP cloud-billing scope.
     () => ({
       extraQueryParams: { access_type: 'offline', p: getConfig().b2cBillingPolicy },
       extraTokenParams: { p: getConfig().b2cBillingPolicy },
     })
   );
 };
 
-export const signIn = async (includeBillingScope = false) => {
+export const signIn = async (includeBillingScope = false): Promise<User> => {
   const args = getSigninArgs(includeBillingScope);
-  const user = await getAuthInstance().signinPopup(args);
-
-  // For B2C record in the auth store whether we requested the GCP cloud-billing scope since there
-  // is no way to determine it after the fact.
-  // For Google we don't need to do this since we can inspect the scope directly in the user object.
-  if (!isGoogleAuthority()) {
-    authStore.update((state) => ({ ...state, hasGcpBillingScopeThroughB2C: includeBillingScope }));
-  }
-  Ajax().Metrics.captureEvent(Events.userLogin, { authProvider: user.profile.idp });
+  const user: User = await getAuthInstance().signinPopup(args);
+
+  const sessionId = uuid();
+  const sessionStartTime: number = Date.now();
+  const userJWT: string = user.id_token!;
+  const decodedJWT: JwtPayload = jwtDecode<JwtPayload>(userJWT);
+  const authTokenCreatedAt: number = (decodedJWT as any).auth_time; // time in seconds when authorization token was created
+  const authTokenExpiresAt: number = user.expires_at!; // time in seconds when authorization token expires
+  const jwtExpiresAt: number = (decodedJWT as any).exp; // time in seconds when jwt expires (should not be read from)
```

Comment:
I updated the comments to attempt to make it more clear. I named them following what they represent rather than where they are sourced from just because I do agree that it is hard to understand. My thinking is to show the intent of the changes because the way the `user` object stores information is a bit unintuitive.